clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesRawResourceProvider.java

```diff
@@ -56,15 +64,59 @@
 
   private Set<KubernetesRawResource> fromRawResourceCacheData(
       Collection<CacheData> rawResourceData) {
-    Set<String> kinds = configuration.getKinds();
-    Set<String> omitKinds = configuration.getOmitKinds();
     return rawResourceData.stream()
         .map(KubernetesRawResource::fromCacheData)
         .filter(Objects::nonNull)
-        .filter(
-            resource ->
-                (kinds.isEmpty() || kinds.contains(resource.getKind().toString()))
-                    && !omitKinds.contains(resource.getKind().toString()))
+        .filter(resource -> includeInResponse(resource))
         .collect(Collectors.toSet());
   }
+
+  private boolean includeInResponse(KubernetesRawResource resource) {
+    Optional<KubernetesCredentials> optionalCredentials =
+        this.accountResolver.getCredentials(resource.getAccount());
+
+    if (!optionalCredentials.isPresent()) {
+      log.warn("Account {} has no credentials", resource.getAccount());
+      return false;
+    }
+
+    KubernetesCredentials credentials = optionalCredentials.get();
+    ImmutableSet<KubernetesKind> omitKinds = credentials.getOmitKinds();
+    ImmutableSet<KubernetesKind> kinds = credentials.getKinds();
+    RawResourcesEndpointConfig epConfig = credentials.getRawResourcesEndpointConfig();
+    List<Pattern> kindPatterns = epConfig.getKindPatterns();
+    List<Pattern> omitKindPatterns = epConfig.getOmitKindPatterns();
+
+    log.debug(
+        "Kinds: {} OmitKinds: {} KindPatterns: {} OmitKindPatterns: {}",
+        kinds.size(),
+        omitKinds.size(),
+        kindPatterns.size(),
+        omitKindPatterns.size());
+
+    // check account level kinds and omitKinds first
+    if (!kinds.isEmpty()) {
+      return kinds.contains(resource.getKind());
```

Comment:
What should happen if `kinds` includes `pod` but that is actually forbidden by `omitKindExpressions`? Shouldn't it only be included if it's allowed by both types of lists in case they are defined?